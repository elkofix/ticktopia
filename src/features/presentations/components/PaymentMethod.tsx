import { CreditCard, Shield, CheckCircle, Lock } from 'lucide-react';

export function PaymentMethods() {
  const creditCards = [
    {
      name: 'Visa',
      color: 'bg-gradient-to-r from-blue-600 to-blue-700',
      textColor: 'text-white',
      logo: 'VISA'
    },
    {
      name: 'Mastercard',
      color: 'bg-gradient-to-r from-red-500 to-orange-500',
      textColor: 'text-white',
      logo: 'MC',
      circles: true
    }
  ];

  return (
    <div className="bg-white p-8 rounded-2xl shadow-lg mb-12 border border-gray-100">
      <div className="flex items-center mb-6">
        <CreditCard className="w-7 h-7 text-blue-600 mr-3" />
        <h2 className="text-3xl font-bold text-gray-900">Métodos de Pago</h2>
      </div>
      
      <div className="bg-blue-50 border border-blue-200 rounded-lg p-4 mb-8">
        <div className="flex items-center">
          <Shield className="w-5 h-5 text-blue-600 mr-2" />
          <p className="text-blue-800 font-medium">
            Aceptamos únicamente tarjetas de crédito para garantizar la seguridad de tu compra
          </p>
        </div>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-3 gap-6 mb-8">
        {creditCards.map((card, index) => (
          <div
            key={index}
            className="group relative overflow-hidden rounded-xl shadow-md hover:shadow-xl transition-all duration-300 hover:-translate-y-2"
          >
            <div className={`${card.color} p-6 h-40 relative`}>
              {/* Chip simulado */}
              <div className="absolute top-4 left-4 w-8 h-6 bg-yellow-400 rounded opacity-80"></div>
              
              {/* Logo de la tarjeta */}
              <div className="absolute top-4 right-4">
                {card.circles ? (
                  <div className="flex items-center">
                    <div className="w-6 h-6 bg-red-500 rounded-full opacity-90"></div>
                    <div className="w-6 h-6 bg-yellow-400 rounded-full -ml-3 opacity-90"></div>
                  </div>
                ) : (
                  <div className={`${card.textColor} font-bold text-lg tracking-wider`}>
                    {card.logo}
                  </div>
                )}
              </div>

              {/* Número de tarjeta simulado */}
              <div className="absolute bottom-8 left-4">
                <div className={`${card.textColor} font-mono text-lg tracking-widest opacity-90`}>
                  •••• •••• •••• ••••
                </div>
              </div>

              {/* Nombre y fecha */}
              <div className="absolute bottom-4 left-4 right-4 flex justify-between">
                <div className={`${card.textColor} text-sm opacity-80`}>
                  NOMBRE APELLIDO
                </div>
                <div className={`${card.textColor} text-sm opacity-80`}>
                  12/28
                </div>
              </div>

              {/* Efecto de brillo */}
              <div className="absolute inset-0 bg-gradient-to-r from-transparent via-white to-transparent opacity-0 group-hover:opacity-10 transform -skew-x-12 group-hover:animate-pulse"></div>
            </div>
            
            <div className="bg-white p-4">
              <div className="flex items-center justify-between">
                <span className="font-semibold text-gray-900">{card.name}</span>
                <CheckCircle className="w-5 h-5 text-green-500" />
              </div>
            </div>
          </div>
        ))}
      </div>

     
    </div>
  );
}