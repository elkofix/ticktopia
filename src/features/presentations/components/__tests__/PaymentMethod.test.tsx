import React from 'react';
import { render, screen } from '@testing-library/react';
import { CreditCard, Shield, CheckCircle } from 'lucide-react';
import '@testing-library/jest-dom';
import { PaymentMethods } from '../PaymentMethod';

// Mock the lucide-react icons
jest.mock('lucide-react', () => ({
  CreditCard: jest.fn(() => <svg data-testid="credit-card-icon" />),
  Shield: jest.fn(() => <svg data-testid="shield-icon" />),
  CheckCircle: jest.fn(() => <svg data-testid="check-circle-icon" />),
  Lock: jest.fn(() => <svg data-testid="lock-icon" />),
}));

describe('PaymentMethods', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the main container correctly', () => {
    render(<PaymentMethods />);
    
    const container = screen.getByTestId('payment-methods-container');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('bg-white');
    expect(container).toHaveClass('p-8');
    expect(container).toHaveClass('rounded-2xl');
    expect(container).toHaveClass('shadow-lg');
    expect(container).toHaveClass('mb-12');
    expect(container).toHaveClass('border');
    expect(container).toHaveClass('border-gray-100');
  });

  it('displays the correct heading with icon', () => {
    render(<PaymentMethods />);
    
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveTextContent('Métodos de Pago');
    expect(heading).toHaveClass('text-3xl');
    expect(heading).toHaveClass('font-bold');
    expect(heading).toHaveClass('text-gray-900');
    
    expect(screen.getByTestId('credit-card-icon')).toBeInTheDocument();
  });

  it('shows the security notice with shield icon', () => {
    render(<PaymentMethods />);
    
    const noticeBox = screen.getByTestId('security-notice');
    expect(noticeBox).toBeInTheDocument();
    expect(noticeBox).toHaveClass('bg-blue-50');
    expect(noticeBox).toHaveClass('border');
    expect(noticeBox).toHaveClass('border-blue-200');
    
    const noticeText = screen.getByText(/Aceptamos únicamente tarjetas de crédito/i);
    expect(noticeText).toBeInTheDocument();
    expect(noticeText).toHaveClass('text-blue-800');
    expect(noticeText).toHaveClass('font-medium');
    
    expect(screen.getByTestId('shield-icon')).toBeInTheDocument();
  });

  it('renders all credit card options', () => {
    render(<PaymentMethods />);
    
    const cards = screen.getAllByTestId('credit-card');
    expect(cards).toHaveLength(2);
    
    // Visa card
    expect(screen.getByText('Visa')).toBeInTheDocument();

  });


  it('matches snapshot', () => {
    const { asFragment } = render(<PaymentMethods />);
    expect(asFragment()).toMatchSnapshot();
  });
});