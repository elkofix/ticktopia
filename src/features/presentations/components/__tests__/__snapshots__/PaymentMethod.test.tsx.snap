// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethods matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-8 rounded-2xl shadow-lg mb-12 border border-gray-100"
    data-testid="payment-methods-container"
  >
    <div
      class="flex items-center mb-6"
    >
      <svg
        data-testid="credit-card-icon"
      />
      <h2
        class="text-3xl font-bold text-gray-900"
      >
        Métodos de Pago
      </h2>
    </div>
    <div
      class="bg-blue-50 border border-blue-200 rounded-lg p-4 mb-8"
      data-testid="security-notice"
    >
      <div
        class="flex items-center"
      >
        <svg
          data-testid="shield-icon"
        />
        <p
          class="text-blue-800 font-medium"
        >
          Aceptamos únicamente tarjetas de crédito para garantizar la seguridad de tu compra
        </p>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-6 mb-8"
    >
      <div
        class="group relative overflow-hidden rounded-xl shadow-md hover:shadow-xl transition-all duration-300 hover:-translate-y-2"
        data-testid="credit-card"
      >
        <div
          class="bg-white p-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <span
              class="font-semibold text-gray-900"
            >
              Visa
            </span>
            <svg
              data-testid="check-circle-icon"
            />
          </div>
        </div>
      </div>
      <div
        class="group relative overflow-hidden rounded-xl shadow-md hover:shadow-xl transition-all duration-300 hover:-translate-y-2"
        data-testid="credit-card"
      >
        <div
          class="bg-white p-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <span
              class="font-semibold text-gray-900"
            >
              Mastercard
            </span>
            <svg
              data-testid="check-circle-icon"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
